### ISSUE TITLE
[OJS] Recoginze https URL to Creative Commons licenses

### ISSUE THREAD

**Issue by zuphilip** [pkp/pkp-lib#4514]
Currently, for the creative commons license the images are shown only if the URL is entered as a `http` URL and without an ending backslash, e.g. for
```
https://creativecommons.org/licenses/by-nc-nd/4.0/
```
you have to enter
```
http://creativecommons.org/licenses/by-nc-nd/4.0
```
I am speaking about the URL you can enter in the production step > schedule for publication:

![ojs-publication-license](https://user-images.githubusercontent.com/5199995/53031383-0adf5980-346d-11e9-87b4-ece99a13e8a6.png)

If you enter here another form of the CC-license, e.g. as a https-URL, then there will be no image shown. Is it possible to recognize in OJS different forms for the CC-licenses, especially the form suggested by CC itself?

**Comment by jamshidhashimi** [comment]
@zuphilip, what version of OJS are you using?

Have you seen the following commits in the Master?
https://github.com/pkp/pkp-lib/commit/e45a1818f1d6dcda7539f226bdae737655d20993#diff-3fed0569f5f0ae40c985f62cfe134c27
https://github.com/pkp/pkp-lib/commit/cdbd845250bdd1f4c49dc69804ecace2bfa04e3a#diff-f69e5caa980791d6835909656e9eace7

cc: @asmecher

**Comment by zuphilip** [comment]
We are using OJS 3.1.1.2. No, I haven't seen these changes. This might solve the `https` issue, but it seems then still that ending backslash will cause problems, when entered.

I would imagine, that several variants of entering a Creative Commons URL should be recognized. Maybe a simple cleaning step is enough to do first, i.e.
  1. Delete the protocol `http` or `https` at the beginning; delete a possible backslash at the end
  2. Check the cleaned string against a list of know licenses

**Comment by asmecher** [comment]
@zuphilip, if you're able to test this change:
https://github.com/pkp/pkp-lib/pull/4539
...and confirm whether it works, I'll commit it for the soon-to-be-released OJS 3.1.2.

**Comment by asmecher** [comment]
Thanks, @zuphilip!
